packages/sanity/src/core/store/events/getDocumentChanges.ts

```diff
@@ -0,0 +1,261 @@
+  events: DocumentGroupEvent[]
+  documentId: string
+}) {
+  const initialValue = incremental.wrap<EventMeta>(omitRev(initialDoc), null)
+  let document = incremental.wrap<EventMeta>(omitRev(initialDoc), null)
+  let finalDocument = omitRev(initialDoc)
+  transactions.forEach((transaction, index) => {
+    const meta: EventMeta = {
+      transactionIndex: index,
+      event: events.find(
+        (event) =>
+          event.type !== 'EditDocumentVersion' &&
+          'revisionId' in event &&
+          event.revisionId === transaction.id,
+      ),
+    }
+    const effect = transaction.effects[documentId]
+    if (effect) {
+      document = incremental.applyPatch(document, effect.apply, meta)
+      finalDocument = applyPatch(finalDocument, effect.apply)
+    }
+  })
+
+  const diff = diffValue({
+    transactions,
+    fromValue: initialValue,
+    fromRaw: initialDoc,
+    toValue: document,
+    toRaw: finalDocument,
+  }) as ObjectDiff
+  return diff
+}
+
+function removeDuplicatedTransactions(transactions: TransactionLogEventWithEffects[]) {
+  const seen = new Set()
+  return transactions.filter((tx) => {
+    if (seen.has(tx.id)) return false
+    seen.add(tx.id)
+    return true
+  })
+}
+
+export function getDocumentChanges({
+  eventsObservable$,
+  documentId,
+  client,
+  to$,
+  since$,
+  remoteTransactions$,
+}: {
+  eventsObservable$: Observable<EventsObservableValue>
+  documentId: string
+  client: SanityClient
+  to$: Observable<EventsStoreRevision | null>
+  remoteTransactions$: Observable<TransactionLogEventWithEffects[]>
+  since$: Observable<EventsStoreRevision | null>
+}): Observable<{loading: boolean; diff: ObjectDiff | null}> {
+  let lastResolvedSince: string | null = null
+  let lastResolvedTo: string | null = null
+  let lastTransactions: TransactionLogEventWithEffects[] = []
+
+  return combineLatest(to$, since$, eventsObservable$).pipe(
+    switchMap(([toObs, since, {events}]) => {
+      const to = toObs?.document
+      let sinceDoc: SanityDocument | undefined = undefined
+      if (since?.document) {
+        sinceDoc = since?.document
+      } else {
+        const selectedToEvent = events.find((event) => event.id === to?._rev)
+        const isShowingCreationEvent =
+          selectedToEvent && isCreateDocumentVersionEvent(selectedToEvent)
+        if (isShowingCreationEvent && to) {
+          sinceDoc = {_type: to._type, _id: to._id, _rev: to._rev} as SanityDocument
+        }
+      }
+      if (!sinceDoc) {
+        return of({loading: false, diff: null})
+      }
+
+      return remoteTransactions$.pipe(
+        switchMap((remoteTx) => {
+          const viewingLatest = !to?._rev
+          const getTransactions = (): Observable<TransactionLogEventWithEffects[]> => {
+            if (viewingLatest && lastResolvedSince === sinceDoc._rev) {
+              // The document has been previously resolved and it's on latest, we can use the remote transactions, we don't need to fetch them again
```

Comment:
Latest might be a context that disappears (or maybe you are even mentioning a different thing than I'm thinking). Might be worth to explicitly say what is the "latest" of that you are thinking this is talking about :)

Reply:
Added extra context to what `viewingLatest` means, thanks for asking :)